Frontend/ui-driver/src/Screens/DriverCompleteProfileScreen/Controller.purs

```diff
@@ -0,0 +1,201 @@
+import Common.Types.App (LazyCheck(..))
+import MerchantConfig.Utils (Merchant(..), getMerchant)
+import Helpers.Utils (getVehicleVariantImage)
+import Language.Strings (getString)
+import Language.Types (STR(..))
+import Data.Array as DA
+import Components.PrimaryButton.Controller as PrimaryButton
+import Engineering.Helpers.Utils as EHU
+import Components.Calendar.Controller as CalendarController
+import Common.Resources.Constants (maxImageUploadInIssueReporting)
+import Effect.Uncurried (runEffectFn3, runEffectFn1, runEffectFn4, runEffectFn5)
+import Services.EndPoints as EndPoint
+import Components.AddImagesModel.Controller as AddImagesModel
+import Components.ViewImageModel as ViewImageModel
+import Engineering.Helpers.Commons (getNewIDWithTag, getCurrentUTC)
+import Effect.Class (liftEffect)
+import Data.TraversableWithIndex (forWithIndex)
+import JBridge (addMediaFile, clearFocus, generatePDF, hideKeyboardOnNavigation, lottieAnimationConfig, removeMediaPlayer, renderBase64ImageFile, saveAudioFile, scrollToEnd, startAudioRecording, startLottieProcess, stopAudioRecording, toast, uploadFile, uploadMultiPartData, openUrlInApp)
+import Data.Int(toNumber, fromString)
+import Components.InputTextView as InputTextView
+import Components.PrimaryButton.Controller as PrimaryButtonController
+import Engineering.Helpers.Commons as EHC
+import Debug(spy)
+
+data Action = OnClickPledge String Boolean 
+            | OnClickVehicalOffer String Boolean 
+            | OnClickLanguages String Boolean 
+            | OnClickWhyNy String Boolean 
+            | OnClickDone PrimaryButton.Action 
+            | TextChanged String String 
+            | ShowCalendarPopup 
+            | CalendarAC CalendarController.Action 
+            | DatePicker String String Int Int Int 
+            | CallBackImageUpload String String String 
+            | ImageUploadCallback  String String String 
+            | UploadMultiPartDataCallback  String String 
+            | AddImagesModelAction AddImagesModel.Action 
+            | ViewImageModelAction ViewImageModel.Action 
+            | OnClickUpload 
+            | OnClickDelete Int
+            | GoBack 
+            | NoAction
+            | OnDateSelect Int Int
+            | Done
+            | InputTextAC InputTextView.Action 
+
+instance showAction :: Show Action where
+  show _ = ""
+instance loggableAction :: Loggable Action where
+  performLog action appId = case action of
+    _ -> pure unit
+
+data ScreenOutput = SubmitRes DriverCompleteProfileScreenState | GoToProfile DriverCompleteProfileScreenState
+
+eval :: Action -> DriverCompleteProfileScreenState -> Eval Action ScreenOutput DriverCompleteProfileScreenState
+
+eval (OnClickPledge title isSelected) state = do
+  if title == "Other" then do
+    if isSelected then continue state { data { inputTextState { others { pledge = "" } } } } else continue state { data { inputTextState { component = "pledge" } }, props { showInputTextView = true } }
```

Comment:
@ROHIT-43 No string comparison, Change it to enum type

Reply:
i think this is the better way in this case because the string i am trying to match (title) also using that string to update the state.